.scrutinizer.yml

```diff
@@ -2,24 +2,42 @@ checks:
     python:
         code_rating: true
         duplicate_code: true
-filter:
-    paths: ['kytos/*', 'tests/*']
-    excluded_paths:
-        - 'kytos/web-ui/*'
 build:
-    environment:
-        python: 3.6
-        postgresql: false
-        redis: false
-    dependencies:
-        before:
-            - pip install coverage
-    tests:
-        override:
-            -
-                command: 'tox'
-                coverage:
-                    file: '.coverage'
-                    config_file: '.coveragerc'
-                    format: 'py-cc'
-            - py-scrutinizer-run
+    nodes:
+        py37:
+            environment:
+                python: 3.7.1
+                postgresql: false
+                redis: false
+            dependencies:
+                before:
+                    - pip install coverage
+            tests:
+                override:
+                    -
+                        command: 'tox -e py37'
+                        only_if: 'tox -a | grep -q py37'
+                        idle_timeout: 300
+                        coverage:
+                            file: '.coverage'
+                            config_file: '.coveragerc'
+                            format: 'py-cc'
+                    - py-scrutinizer-run
+        pyOthers:
+            environment:
+                python: 3.6
+                postgresql: false
+                redis: false
+            dependencies:
+                before:
+                    - pip install coverage
+            tests:
+                override:
+                    -
+                        command: 'for env in $(tox -a | grep -v py37); do tox -e $env; done'
```

Comment:
@italovalcy wouldn't this cause a mismatch between scrutinizer node running the environment on `3.6` and then running tox for these three other versions?

```
py36
py38
py39
 ```
 
I wonder if we should each version of a different environment like we have on tox envlist, if positive, would they run in parallel?

Reply:
@viniarck the nodes are not executed in parallel. There will be no mismatch, because the scrutinizer node `python: 3.6` does support tox environments py36, py38 and py39 (more info here: https://github.com/kytos-ng/kytos/issues/146#issuecomment-988172068, which also includes a log of a execution). This config is currently in production for Kytos-ng/kytos with some inspections already done: https://scrutinizer-ci.com/g/kytos-ng/kytos/inspections

Reply:
Got it. Thanks for clarifying @italovalcy. It's really great to have scrutinizer and its reporting capabilities, good stuff.